lib/carbonite/migrations.ex

```diff
@@ -98,9 +98,10 @@ defmodule Carbonite.Migrations do
     carbonite_prefix = Keyword.get(opts, :carbonite_prefix, default_prefix())
 
     """
-    CREATE TRIGGER capture_changes_into_#{carbonite_prefix}_trigger
+    CREATE CONSTRAINT TRIGGER capture_changes_into_#{carbonite_prefix}_trigger
```

Comment:
👏 Nice move, deferring the trigger 💚. Wouldn't it make sense to set `deferred` to be the new default? Or is that not an option, since rolling back because of the trigger can still happen and have unintended consequences (because of transation side-effects)?

Reply:
Idk, considered it but figured the current behaviour of failing early when no transaction is inserted is still what we / most people want? TBH I think for most people it won't matter, so I kept the current way.

---

The `DEFERRABLE` by itself is of course a change in behaviour. If somebody has code that runs `SET CONSTRAINTS ALL DEFERRED` for whatever reason, it would now also defer the carbonite trigger. Buuut only for "new" triggers, i.e. created with this patch. And generally the `SET CONSTRAINTS` thing seems unlikely, so :shrug: